### DISCUSSION TITLE
Register-PnPManagementShellAccess  failure

### DISCUSSION THREAD

**Discussion by figuerres** [pnp/powershell#1213] [Q&A] [answered]
i read the documentation but i am lost here...
my login has sharepoint admin rights so what other roles do i need to have for this command to work ?

"You need to run this cmdlet with an identity that has write access to the Azure AD." 
what does that mean ?   

https://pnp.github.io/powershell/articles/authentication.html

i am trying to setup an app that we need to have upload files to our sharepoint site. i need to setup the app so that it can run automatically.  but just getting the right access to create the app security seems to need me to be an ad admin ....

**Comment by ToddKlindt** [comment]
It's a complicated bit, but fortunately you only have to do it once. You have to run `Register-PnPManagementShellAccess` _once_ as a Tenant Admin. That sets up an Azure Application in Azure AD that the PnP commands use. There's no way around it, unfortunately.

**Reply by figuerres** [reply]
Possibly I was not clear. I try to run that command and it tells me I need to use an account with other permissions. I am asking what role my account needs for the command to work. The documentation says something about a right to write to ad. What role should my account have, name the role in the documentation ad has many options, I don't want to guess and I don't want more than is required.

**Reply by figuerres** [reply]
https://pnp.github.io/powershell/articles/connecting.html
says:
Connect by using the PnP Management Shell Multi-Tenant Azure AD Application
You will have to consent / register the PnP Management Shell Multi-Tenant Azure AD Application in your own tenant:

Register-PnPManagementShellAccess
This will launch a device login flow that will ask you to consent to the application. Notice that is only required -once- per tenant. You will need to have appropriate access rights to be able to consent applications in your Azure AD.

butwhat are the right that i need to have ?? please document what they are !

**Reply by figuerres** [reply]
is this the missing setting ?
https://sharepoint-tricks.com/the-required-office-365-role-to-run-pnp-powershell-with-scopes/
Application admin

please document that if it is the one !

**Comment by ToddKlindt** [comment, marked as answer]
The official role name is "Global Administrator." Your tenant has one of those. You only need to log in as that user and run that command once, then the PnP.PowerShell module will work for anyone in that tenant.
![image](https://user-images.githubusercontent.com/2659747/136244750-10ab71ce-1e41-48bd-8f49-4b79ea94496e.png)

**Comment by figuerres** [comment]
thanks, so far this has been a list of roles needed to just get to running the commands to make the setting for my app....
i am documenting this locally for our shop in case we need it later..... but it will be good to get this into the official docs for anyone else going into do a sharepoint app first time.
